rust/xaynet-analytics/src/database/controller_data/data_model.rs

```diff
@@ -15,23 +15,22 @@ impl ControllerData {
     }
 }
 
-impl IsarAdapter for ControllerData {
-    fn into_field_properties() -> IntoIter<FieldProperty> {
-        vec![FieldProperty::new(
-            "time_data_sent".to_string(),
-            DataType::String,
-        )]
-        .into_iter()
-    }
+impl<'ctrl> TryFrom<ControllerDataAdapter> for ControllerData {
+    type Error = anyhow::Error;
 
-    fn write_with_object_builder(&self, object_builder: &mut ObjectBuilder) {
-        object_builder.write_string(Some(&self.time_data_sent.to_rfc3339()));
+    fn try_from(adapter: ControllerDataAdapter) -> Result<Self, Self::Error> {
+        Ok(ControllerData::new(
+            DateTime::parse_from_rfc3339(&adapter.time_data_sent)
+                .unwrap()
```

Comment:
Can we return the error instead of using `unwrap`? Or can you add a comment on why it would be safe to use `unwrap`?

Reply:
sounds good